scripts/irods/controller.py

```diff
@@ -223,8 +223,16 @@ def wait_for_server_to_stop(self, retry_count=100):
         l = logging.getLogger(__name__)
         for attempt in range(retry_count):
             l.debug('Waiting for iRODS server to shut down. Attempt #%s', attempt)
-            if self.get_server_pid() is None:
+            server_pid = self.get_server_pid()
+            if server_pid is None:
                 return
+
+            # checking for zombieness
+            proc = psutil.Process(server_pid)
+            if proc.status() == psutil.STATUS_ZOMBIE:
+                l.warning('Server process was terminated, but is now a zombie.  Your environment is not reaping them.')
```

Comment:
Sometimes warning messages pose a question to provide help on what might be causing a problem. Maybe something like...

"Terminating server process resulted in a zombie process. Is it running as the init process?"

Reply:
Any of the suggestions other than mine seem fine to me.

Reply:
No preference.

My proposals are mostly about pointing out that the target PID comes from the PID file (in /var/run/irods).

Notice the script doesn't know when the admin launched the server with a custom PID file. I don't think this is an issue since the controller.py logic isn't in the main path for shutdown. The common/expected way to shutdown the server is through the `kill` command or the service manager (if configured for that).

Reply:
updated the words...

```
Terminating the iRODS server referenced by the iRODS PID file resulted in a zombie process.  Consider running an init-like process as PID 1. 
```

Reply:
Message looks good to me.

Resolving.